packages/console/src/components/CodeEditor/index.tsx

```diff
@@ -12,11 +12,44 @@ type Props = {
   isReadonly?: boolean;
   value?: string;
   onChange?: (value: string) => void;
+  tabSize?: number;
 };
 
-const CodeEditor = ({ className, language, isReadonly = false, value = '', onChange }: Props) => {
+const CodeEditor = ({
+  className,
+  language,
+  isReadonly = false,
+  value = '',
+  onChange,
+  tabSize = 4,
+}: Props) => {
+  const [text, setText] = useState<{ caret: number; target?: HTMLTextAreaElement }>({
+    caret: -1,
+    target: undefined,
+  });
+
+  useEffect(() => {
+    if (text.caret >= 0) {
+      text.target?.setSelectionRange(text.caret + tabSize, text.caret + tabSize);
+    }
+  }, [tabSize, text]);
```

Comment:
This `useEffect` is not needed IMO.

Reply:
Since `setSelectionRange` has to be called after the state (or value of the `textarea`), I decided to use `useEffect` but I changed it better way.